### PR TITLE
Add clique method for robustness verification of decision tree ensembles/gradient boosted decision trees

### LINKED ISSUE #124: Add robustness verification/metric for tree-based classifiers
Paper links:
- https://arxiv.org/abs/1906.03849
- https://arxiv.org/abs/1906.03526

### PR DESCRIPTION
**Pull request by beat-buesser**
# Description

This pull request adds the clique verification for robustness verification of decision tree ensembles/gradient boosted decision trees

Fixes #124 

## Type of change

Please check all relevant options.

- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by minhitbk** [comment]
The code looks good to me. I just have one minor comment, on lines 277-282: 

        if norm == 0:
            resulting_distance = resulting_distance
        elif norm == np.inf:
            resulting_distance = resulting_distance
        else:
            resulting_distance = pow(resulting_distance, 1.0 / norm)

Could we change this code to:

        if norm == np.inf:
            resulting_distance = resulting_distance
        elif norm > 0:
            resulting_distance = pow(resulting_distance, 1.0 / norm)

**Comment by beat-buesser** [comment]
@minhitbk Thank you very much for the review!
